tests/core/engine_adapter/test_mixins.py

```diff
@@ -54,32 +54,35 @@ def test_logical_replace_query_does_not_exist(
     make_mocked_engine_adapter: t.Callable, mocker: MockerFixture, make_temp_table_name: t.Callable
 ):
-    adapter = make_mocked_engine_adapter(LogicalReplaceQueryMixin, "postgres")
-    adapter.cursor.fetchone.return_value = (1,)
-
-    temp_table_mock = mocker.patch("sqlmesh.core.engine_adapter.EngineAdapter._get_temp_table")
-    table_name = "db.table"
-    temp_table_id = "abcdefgh"
-    temp_table_mock.return_value = make_temp_table_name(table_name, temp_table_id)
-
-    mocker.patch(
-        "sqlmesh.core.engine_adapter.postgres.LogicalReplaceQueryMixin.table_exists",
-        return_value=True,
-    )
-    mocker.patch(
-        "sqlmesh.core.engine_adapter.postgres.LogicalReplaceQueryMixin.columns",
-        return_value={"col": exp.DataType(this=exp.DataType.Type.INT)},
-    )
-
-    adapter.replace_query(table_name, parse_one(f"SELECT col + 1 AS col FROM {table_name}"))
-
-    assert to_sql_calls(adapter) == [
-        f'CREATE SCHEMA IF NOT EXISTS "db"',
-        f'CREATE TABLE IF NOT EXISTS "db"."__temp_table_{temp_table_id}" AS SELECT "col" FROM "db"."table"',
-        'TRUNCATE "db"."table"',
-        f'INSERT INTO "db"."table" ("col") SELECT "col" + 1 AS "col" FROM "db"."__temp_table_{temp_table_id}"',
-        f'DROP TABLE IF EXISTS "db"."__temp_table_{temp_table_id}"',
-    ]
+    for dialect in ["postgres", "spark"]:
+        id_char = '"' if dialect == "postgres" else "`"
+
+        adapter = make_mocked_engine_adapter(LogicalReplaceQueryMixin, dialect)
+        adapter.cursor.fetchone.return_value = (1,)
+
+        temp_table_mock = mocker.patch("sqlmesh.core.engine_adapter.EngineAdapter._get_temp_table")
+        table_name = "db.table"
+        temp_table_id = "abcdefgh"
+        temp_table_mock.return_value = make_temp_table_name(table_name, temp_table_id)
+
+        mocker.patch(
+            "sqlmesh.core.engine_adapter.postgres.LogicalReplaceQueryMixin.table_exists",
+            return_value=True,
+        )
+        mocker.patch(
+            "sqlmesh.core.engine_adapter.postgres.LogicalReplaceQueryMixin.columns",
+            return_value={"col": exp.DataType(this=exp.DataType.Type.INT)},
+        )
+
+        adapter.replace_query(table_name, parse_one(f"SELECT col + 1 AS col FROM {table_name}"))
+
+        assert to_sql_calls(adapter) == [
+            f"CREATE SCHEMA IF NOT EXISTS {id_char}db{id_char}",
+            f"CREATE TABLE IF NOT EXISTS {id_char}db{id_char}.{id_char}__temp_table_{temp_table_id}{id_char} AS SELECT {id_char}col{id_char} FROM {id_char}db{id_char}.{id_char}table{id_char}",
+            f"TRUNCATE {id_char}db{id_char}.{id_char}table{id_char}",
+            f"INSERT INTO {id_char}db{id_char}.{id_char}table{id_char} ({id_char}col{id_char}) SELECT {id_char}col{id_char} + 1 AS {id_char}col{id_char} FROM {id_char}db{id_char}.{id_char}__temp_table_{temp_table_id}{id_char}",
+            f"DROP TABLE IF EXISTS {id_char}db{id_char}.{id_char}__temp_table_{temp_table_id}{id_char}",
```

Comment:
What do you think about leaving this test as is and then adding a new `replace_query` test to Spark that has a select reference query and then verify that it then outputs SQL in the above pattern? That way when I look at the Spark tests I would see tests for regular replace_where query, replace_where dataframe, and then replce_where self-reference: https://github.com/TobikoData/sqlmesh/blob/e2f498bfc99fe909663492e686e19e00eef9468d/tests/core/engine_adapter/test_spark.py#L108-L129